### PR TITLE
Add user profile page, and a few refactors

### PR DESCRIPTION
**Pull request by tennantje**
What's new?
- User route for show and destroy
- User profile with gravatar and display of users submitted gifs
- Refactoring of session related stuff
- Refactoring of some view stuff (partials)

### CODE REVIEW

**Comment by keithpitt** [comment]
Screenshots!

**Comment by keithpitt** [comment]
Awesome btw :)

**Comment by tennantje** [comment]
Cheers dude,
I'll sort out some screenshots when I get home this evening.

Travis CI Build failed?

**Comment by keithpitt** [comment]
Yeah, that's odd. Do the tests run locally?

**Comment by tennantje** [comment]
screen shot 2013-07-17 at 5 34 35 pm 2
screen shot 2013-07-17 at 5 34 55 pm 2

**Comment by tennantje** [comment]
turns out the tests were failing, because the user wasn't signed in when trying to access their show-profile page. I'll maybe look to sort out the tests in a future commit.

**Comment by keithpitt** [comment]
Looks like theres a merge conflict with master. If you merge master in, I'll merge this bad boy!

**Comment by keithpitt** [comment]
Can you remove the "destroy account" feature? I'm not really sure if we should allow that yet. I found a security whole with your implementation that I'll just comment now.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  private
+
+    def redirect_unless_authorised
+      redirect_to root_path unless user_signed_in? && current_user?(current_resource)
+    end
+
+    def current_resource
+      @current_resource ||= User.find(params[:id]) if params[:id]
```
Comment (keithpitt) [inline app/controllers/users_controller.rb:16]: Security hole!

**Comment by tennantje** [comment]
Believe this is now resolved Mr Keith!

**Comment by keithpitt** [comment]
Don't forget to remove the "delete account" feature :)

**Comment by tennantje** [comment]
Sorted, let me know if that's correct?

**Pull request merged**